locales/en/messages.json

```diff
@@ -592,7 +592,11 @@
         "message": "<p><strong>Betaflight</strong> is a flight controller software that is <strong>open source</strong> and is available free of charge <strong>without warranty</strong> to all users.</p><p>If you found the Betaflight or Betaflight configurator useful, please consider <strong>supporting</strong> its development by donating.</p>"
     },
     "defaultDonateBottom": {
-        "message": "<p>If you want to contribute financially on an ongoing basis, you should consider becoming a patron for us on <a href='https://www.patreon.com/betaflight'>Patreon</a>.</p>"
+        "message": "<p>If you want to contribute financially on an ongoing basis, you should consider becoming a patron for us on $t(patreonLink.message).</p>"
+    },
+    "patreonLink": {
+        "message": "<a href=\"https://www.patreon.com/betaflight\" target=\"_blank\">Patreon</a>",
+        "description": "Patreon is name, and should not require translation"
```

Comment:
If the Patreon word and URL must not be translated is better to take it out of the translations file. Some user can translate them without seeing the description and we will not be aware of that if we no look specifically at that element.
I think is better to have them in the js file and pass them as parameter to the original phrase.

Reply:
@mikeller what do you think? I have seen that you approved it, so I'm not sure if my suggestion is the right way.

Reply:
My knowledge of the localization stuff is quite limited, I saw that the string was tagged via an html tag, instead of the usual javascript call, and assumed I had to do everything within the json file.

https://github.com/betaflight/betaflight-configurator/blob/0b035c1ead60d6b2cec4aa269d695103c84931a4/src/tabs/landing.html#L32

Reply:
True, in this case we don't have the possibility to pass parameters 😭. We can:
- Modify the i18n code to support it
- Move the translation from html to js
- Let as is

Reply:
I guess if we could add the resource before the i18n stuff fires off? 

https://github.com/betaflight/betaflight-configurator/blob/0b035c1ead60d6b2cec4aa269d695103c84931a4/src/js/LogoManager.js#L102-L106

https://github.com/betaflight/betaflight-configurator/blob/0b035c1ead60d6b2cec4aa269d695103c84931a4/locales/en/messages.json#L3751-L3753

Reply:
This is a way to do it, but I don't like it too much neither, it adds hidden elements to the translations file but in this case (a Patreon link) is not as ugly as in the case you are pointing.

The other option is to add a `i18nparams="xxxxxx"` attribute, and modify here: https://github.com/betaflight/betaflight-configurator/blob/0b035c1ead60d6b2cec4aa269d695103c84931a4/src/js/localization.js#L82

to make use of it. It will be a little ugly too, because we add the link and other elements to an HTML attribute.

Let's see what @mikeller think, maybe he prefers your initial solution :)

Reply:
Another solution, I haven't tested it, is to split the translations file into two: one with constants, an the other with the phrases to translate. In this case it will be cleaner and we will need only to add the  `messages_constants.json` file to the initialization of the i18n system. This file will not be uploaded to the translators, and we can add other phrases that we don't want to translate.

Reply:
Not sure what is best to do here.
While I agree that the originally proposed solution is not 'pretty', I also think that it is probably the best we can have:
- some translations will contain constants, or will contain other translations (e.g. button labels referenced in help texts). Hence splitting the translations into 'constants' and non-constants, and injecting the constants into the other translations will lead to a half-half solution;
- we want to be able to inject translations directly into the html, without having to write custom JS for any of those translations, even if they contain constants;
- the above point, and the fact that we cannot define constants in html, kind of point towards needing a solution like what was proposed.

Potentially one refinement we could do is to use a keyword like `DO_NOT_TRANSLATE` in the description of constants (or use a custom property to encode this), and have the build fail if any of the translation files contains a translation for one of these.

Reply:
Having the translations file divided into constants and not constants, does not break anything of the features we want: we can use and reference them from the HTML and from other translations like we do now, in a totally transparent manner.

Reply:
I'm not saying it is going to break anything - I am just not sure that this is a good way to solve this by adding complexity.

Reply:
What I understood in your comment, is that splitting the file into two, needs to change the way that we use to include translations/constants in the HTML/JS. That's not true. The i18next localization that we use will search in both files in a transparent manner for the user if configured correctly. So the developer will use the same instructions/code than before, not matter if it is a translation or a constant, so it does not increase the complexity in any way.

The only drawback, is that the developer must be aware that there are two files, and include the messages in the correct one when adding. Nothing more.

Reply:
That still adds complexity over what we currently have - my suggestion is to keep non-translated and translated strings in the same file, as we currently do, and use some form of indicator to mark not-to-be-translated strings as such, and make it part of the CI build job (or potentially even add hooks to crowdin) to check that these constraints are not violated.

Reply:
Mergeing this, we can still finish the discussion.

Reply:
I have hidden the Patreon link string in Crowdin, but this is not a perfect solution. The users can show the hidden strings (all the languages strings were hidden and we now some users ended translating some of them).